CONTRIBUTING.md

```diff
@@ -5,70 +5,107 @@ If you have any questions or suggestions we are happy to hear them.
+to the extension point.
+
+There are multiple providers that are each responsible in defining which annotation(s) to scan for.
+
+Examples:
+
+| Class | Annotations |
+|-------|-------------|
+| [MicroProfileConfigPropertyProvider](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileConfigPropertyProvider.java) | org.eclipse.microprofile.config.inject.ConfigProperty |
+| [MicroProfileRegisterRestClientProvider](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/core/providers/MicroProfileRegisterRestClientProvider.java) | org.eclipse.microprofile.rest.client.inject.RegisterRestClient |
+| [QuarkusConfigPropertiesProvider](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/QuarkusConfigPropertiesProvider.java) | io.quarkus.arc.config.ConfigProperties |
+| [QuarkusConfigRootProvider](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/QuarkusConfigRootProvider.java) | io.quarkus.runtime.annotations.ConfigRoot |
+| [QuarkusKubernetesProvider](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.jdt/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/QuarkusKubernetesProvider.java) | io.dekorate.kubernetes.annotation.KubernetesApplication<br>io.dekorate.openshift.annotation.OpenshiftApplication<br>io.dekorate.s2i.annotation.S2iBuild<br>io.dekorate.docker.annotation.DockerBuild |
+
+
+## Searching for properties in JARs not in the user's Quarkus project's classpath
+Some Quarkus properties like `quarkus.hibernate-orm.dialect` are not defined in JARs belonging
+to the current Quarkus project's classpath. Some Quarkus classpath JARs have deployment JARs.
+Generally speaking, deployment JARs are not present in the project's classpath.
 
-The scanner will collect the [io.quarkus.deployment.ApplicationConfig](https://github.com/quarkusio/quarkus/blob/master/core/deployment/src/main/java/io/quarkus/deployment/ApplicationConfig.java) class annotated with `@ConfigRoot` and will generate a Quarkus property for each field:
+If classpath JARs have a deployment JAR, it will be defined in the 
+`META-INF/quarkus-extension.properties` file of the classpath JAR.
 
- * `quarkus.application.name` for the `name` field.
- * `quarkus.application.version` for the `version` field.
- 
-In addition, the Quarkus jdt.ls extension is able to manage `Quarkus deployment JAR`. 
+For example, the `quarkus-amazon-dynamodb-1.1.1.Final.jar` has a deployment JAR with
```

Comment:
I suggest you use sample with `quarkus.hibernate-orm` (because other explanation is with this artifcat). It wll require to update the `images/deployment.png`